packages/shorebird_tests/test/ios_test.dart

```diff
@@ -43,7 +43,53 @@ void main() {
         );
       });
 
-      // TODO(erickzanardo): Add tests for flavors.
+      group('when building with a flavor', () {
+        testWithShorebirdProject(
+          'correctly changes the app id',
+          (projectDirectory) async {
+            await projectDirectory.addProjectFlavors();
+            projectDirectory.addShorebirdFlavors();
+
+            await projectDirectory.runFlutterBuildIos(flavor: 'internal');
+
+            final generatedYaml =
+                await projectDirectory.getGeneratedIosShorebirdYaml(
+                    //flavor: 'internal',
+                    );
+
+            expect(generatedYaml['app_id'], equals('internal_123'));
+          },
+        );
+
+        group('when public key passed through environment variable', () {
+          testWithShorebirdProject(
+            'correctly changes the app id and adds the public key',
+            (projectDirectory) async {
+              const base64PublicKey = 'public_123';
+              await projectDirectory.addProjectFlavors();
+              projectDirectory.addShorebirdFlavors();
+
+              await projectDirectory.runFlutterBuildIos(
+                flavor: 'internal',
+                environment: {
+                  'SHOREBIRD_PUBLIC_KEY': base64PublicKey,
+                },
+              );
+
+              final generatedYaml =
+                  await projectDirectory.getGeneratedIosShorebirdYaml(
+                      //flavor: 'internal',
```

Comment:
Did you want this line still?

Reply:
I did not! I used the opportunity to test and confirm that the shorebird.yaml will always end up in the same place in the build folder, no matter which flavor were used.

Good catch!